### ISSUE TITLE
Performance loss when multiple requests

### ISSUE THREAD

**Issue by augustosnk12** [hgourvest/node-firebird#289]
Hello there, 

I'm facing some issue here: I have an endpoint that receives multiples requests from a lot of users. All of it is connected to a unique database. 

The point is when a lot of users are send requests at the same time I can see some delay / loss of performance in the requests. 

**Does this lib can cause this or my issue is another thing?**

**Comment by FabianoCastro** [comment]
Hi, do you are using await?

I used, and very slow, I changed and solved.

**Comment by augustosnk12** [comment]
@FabianoCastro you changed in order to use what?

**Comment by FabianoCastro** [comment]
Like this

returnData(aSQL:string) => {
return new Promise<any[]>(async (res, rej) => {
aDataBase.query(aSQL, [], async (aError: any, aRows: any[]) => {
   if (aErroQr) {
     rej(aErroQr);
   } else {
     res(aRows);
   }
});})
}

**Comment by FabianoCastro** [comment]
I can see in exemple too

db.query('INSERT INTO USERS (ID, ALIAS, CREATED) VALUES(?, ?, ?) RETURNING ID', [1, 'Pe\'ter', new Date()], function(err, result) {
        console.log(result[0].id);
        db.query('SELECT * FROM USERS WHERE Alias=?', ['Peter'], function(err, result) {
            console.log(result);
            db.detach();
        });
    });


https://github.com/hgourvest/node-firebird

**Comment by augustosnk12** [comment]
I see it...going to try it 😉

**Comment by mariuz** [comment]
Expected with a single connection: one Firebird attachment processes one request at a time, so concurrent users queue behind each other. The shipped answer is the connection pool — plus, nowadays, first-class Express guidance:

```js
const pool = Firebird.pool(10, options); // one pool at app startup
app.get('/route', (req, res) => pool.withConnection(db => db.queryAsync(...)) ...);
```

See [README § Using node-firebird with Express.js](https://github.com/hgourvest/node-firebird#using-node-firebird-with-expressjs) (request lifecycle, always-release patterns) and [§ Pool events and metrics](https://github.com/hgourvest/node-firebird#pool-events-and-metrics) for sizing/observability (`waitingCount` tells you when the pool is the bottleneck). Closing as answered.
